nixos/modules/virtualisation/google-compute-config.nix

```diff
@@ -69,6 +69,69 @@ in
   # GC has 1460 MTU
   networking.interfaces.eth0.mtu = 1460;
 
+  systemd.services.fetch-ssh-keys = {
+    description = "Fetch host keys and authorized_keys for root user";
+
+    wantedBy = [ "sshd.service" ];
+    before = [ "sshd.service" ];
+    after = [ "network-online.target" ];
+    wants = [ "network-online.target" ];
+
+    script =
+      let
+        wget = "${pkgs.wget}/bin/wget --retry-connrefused -t 15 --waitretry=10 --header='Metadata-Flavor: Google'";
```

Comment:
I'm feeling a bit uncomforable with us writing our own bashscript invoking wget with some amounts of retries. Is there no native script provided by google/through cloud-init/afterburn that takes care of the fetching and readyness signalling?

Reply:
According to https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying, you're supposed to fetch it manually. The cloud sdk doesn't seem to have any nice helpers for this either :/